test/t8_geometry/t8_gtest_geometry.cxx

```diff
@@ -0,0 +1,296 @@
+ * has the correct name and dimension. */
+TEST_P (geometry, geometry_linear)
+{
+  t8_geometry_linear  linear_geom (dim);
+  char                name[BUFSIZ];
+  snprintf (name, BUFSIZ, "t8_geom_linear_%i", dim);
+  ASSERT_FALSE (strcmp (linear_geom.t8_geom_get_name (), name)) <<
+    "Linear geometry of dim " << dim << "has wrong name. Expected " << name <<
+    " got " << linear_geom.t8_geom_get_name ();
+  ASSERT_EQ (dim,
+             linear_geom.t8_geom_get_dimension ()) <<
+    "Linear geometry of dim " << dim << "has wrong dimension: " <<
+    linear_geom.t8_geom_get_dimension () << ".";
+}
+
+/* Check that the zero geometry for dimensions 0,1,2,3
+ * has the correct name and dimension. */
+TEST_P (geometry, geometry_zero)
+{
+  t8_geometry_zero    zero_geom (dim);
+  char                name[BUFSIZ];
+  snprintf (name, BUFSIZ, "t8_geom_zero_%i", dim);
+  ASSERT_FALSE (strcmp (zero_geom.t8_geom_get_name (), name)) <<
+    "Linear geometry of dim " << dim << "has wrong name. Expected " << name <<
+    " got " << zero_geom.t8_geom_get_name ();
+  ASSERT_EQ (dim,
+             zero_geom.t8_geom_get_dimension ()) << "Linear geometry of dim "
+    << dim << "has wrong dimension: " << zero_geom.t8_geom_get_dimension () <<
+    ".";
+}
+
+/* Check whether the linear geometry map is correct.
+ * We create a cmesh of one tree and unit point/line/square/cube
+ * geometry. We then create random points in a reference tree and
+ * check whether the evaluation is correct. */
+TEST_P (geometry, cmesh_geometry_linear)
+{
+  /* TODO: Add a test for the jacobian, as soon as its implemented. */
+
+  t8_debugf ("Testing linear geometry evaluation.\n");
+
+  /* Create random points in [0,1]^d and check if they are mapped correctly. */
+  t8_geometry_linear  linear_geom (dim);
+  t8_cmesh_t          cmesh;
+
+  int                 num_points = 10000;
+  double              point[3];
+  double              point_mapped[3];
+  int                 seed = 0; /* RNG seed */
+  t8_eclass_t         tree_class;
+  const t8_geometry_c *cmesh_geom;
+
+  /* Build a one tree cmesh on the unit square with linear geometry. */
+  switch (dim) {
+  case 0:
+    tree_class = T8_ECLASS_VERTEX;
+    break;
+  case 1:
+    tree_class = T8_ECLASS_LINE;
+    break;
+  case 2:
+    tree_class = T8_ECLASS_QUAD;
+    break;
+  case 3:
+    tree_class = T8_ECLASS_HEX;
+    break;
+  default:
+    SC_ABORT_NOT_REACHED ();
+  }
+  cmesh = t8_cmesh_new_hypercube (tree_class, sc_MPI_COMM_WORLD, 0, 0, 0);
+
+  /* Double check that the geometry is the linear geometry. */
+  cmesh_geom = t8_cmesh_get_tree_geometry (cmesh, 0);
+  ASSERT_TRUE (cmesh_geom != NULL) << "Could not get cmesh's geometry.";
+  int                 has_same_name = strcmp (cmesh_geom->t8_geom_get_name (),
+                                              linear_geom.t8_geom_get_name
+                                              ());
+  ASSERT_FALSE (has_same_name) <<
```

Comment:
I would improve code readibility here. See the line above, where `strcmp` is called. `strcmp` compares two strings and when they match, the function returns `0`. A zero in C/C++ is per convention a `false` value. That is why the assertion checks for `false` here. This is the correct behavior but difficult to read for programmers. I suggest you change the line as follows:
```suggestion
  ASSERT_EQ (has_same_name, 0) <<
```